src/SearchEngine.java

```diff
@@ -0,0 +1,111 @@
+import strategies.SearchStrategy;
+import strategies.StrategyALL;
+import strategies.StrategyANY;
+import strategies.StrategyNONE;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class SearchEngine {
+
+    Map<String, ArrayList<Integer>> map = new HashMap<>();
+    ArrayList<String> list = new ArrayList<>();
+    static Scanner scanner = new Scanner(System.in);
```

Comment:
Are you sure about the `default` access modifier?

Reply:
yes, I'm sure

Reply:
I still don't understand why these fields have the default access modifier. Can you explain?)

Reply:
this fields need to be a "private final" - my bad.
Now this class is changed)